File: src/main/java/hr/tvz/susac/java/web/flight/model/Flight.java
package hr.tvz.susac.java.web.flight.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "flight_table", schema = "flight_finder")
public class Flight
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", unique = true, nullable = false)
    private Long id;

    @Column(name = "origin_location_code", nullable = false)
    private String originLocationCode;

    @Column(name = "destination_location_code", nullable = false)
    private String destinationLocationCode;

    @Column(name = "departure_date", nullable = false)
    private LocalDate departureDate;

    @Column(name = "return_date")
    private LocalDate returnDate;

    @Column(name = "passenger_count", nullable = false)
    private Integer passengerCount;

    @Column(name = "transfer_count", nullable = false)
    private Integer transferCount;

    @Column(name = "currency_code", nullable = false)
    private String currencyCode;

    @Column(name = "total_price", nullable = false)
    private Double totalPrice;

    @CreationTimestamp
    @Column(name = "date_time_created", nullable = false)
    private LocalDateTime dateTimeCreated;

    @ToString.Exclude
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: web/src/app/models/flight.ts
export interface Flight {
    id: number,
    originLocationCode: string;
    destinationLocationCode: string;
    departureDate: Date,
    returnDate?: Date | null,
    passengerCount: number,
    transferCount: number,
    currencyCode: string,
    totalPrice: number
}

export class Flight implements Flight {
    constructor(init?: FlightFormValues) {
        Object.assign(this, init);
    }
}

export class FlightFormValues {
    originLocationCode: string = '';
    destinationLocationCode: string = '';
    departureDate: Date | null = null;
    returnDate?: Date | null = null;
    passengerCount: number = 0;
    transferCount: number = 0
    currencyCode: string = '';
    totalPrice: number = 0;

    constructor(flight: Flight) {
        this.originLocationCode = flight.originLocationCode;
        this.destinationLocationCode = flight.destinationLocationCode;
        this.departureDate = flight.departureDate;
        this.returnDate = flight.returnDate;
        this.passengerCount = flight.passengerCount;
        this.transferCount = flight.transferCount;
        this.currencyCode = flight.currencyCode;
        this.totalPrice = flight.totalPrice;
    }
}

export interface FlightSearchFormValues {
    originLocationCode: string;
    destinationLocationCode: string;
    departureDate: Date | null;
    returnDate: Date | null;
    passengerCount: number;
    currencyCode: string;
}